chihuahua/src/hcall/wasmtime.rs

```diff
@@ -602,6 +622,24 @@ impl Chihuahua for DummyWasmtimeHostProvisioningState {
             .clone()
     }
 
+    #[inline]
+    fn get_current_stream_source_count(&self) -> usize {
+        HOST_PROVISIONING_STATE
+            .lock()
+            .expect("Failed to obtain lock on host provisioning state.")
+            .get_current_stream_source_count()
+            .clone()
+    }
+
+    #[inline]
+    fn get_expected_stream_sources(&self) -> Vec<u64> {
```

Comment:
Are these clones required?  From memory there may be some weirdness with the `Mutex` which means it's easier to clone, but if not they should be removed in favour of returning borrowed data.

Reply:
It is the weirdness of Mutex. I believe any reference to a mutex becomes invalid once the mutex is released.